t/plugin/ua-restriction.t

```diff
@@ -0,0 +1,584 @@
+GET /hello
+--- more_headers
+User-Agent:Baiduspider/1.0
+--- error_code: 403
+--- response_body
+{"message":"Do you want to do something bad?"}
+
+
+
+=== TEST 22: test remove ua-restriction part 1, enable
+--- config
+    location /enable {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "uri": "/hello",
+                        "upstream": {
+                            "type": "roundrobin",
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            }
+                        },
+                        "plugins": {
+                            "ua-restriction": {
+                                "denylist": [
+                                     "(Baiduspider)/(\\d+)\\.(\\d+)"
+                                ]
+                            }
+                        }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /enable
+--- error_code: 200
+
+
+
+=== TEST 23: test remove ua-restriction part 2
+--- request
+GET /hello
+--- more_headers
+User-Agent:Baiduspider/1.0
+--- error_code: 403
+
+
+
+=== TEST 24: test remove ua-restriction part 3, remove plugin
+--- config
+    location /disable {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "uri": "/hello",
+                        "upstream": {
+                            "type": "roundrobin",
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            }
+                        },
+                        "plugins": {
+                        }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /disable
+--- error_code: 200
+
+
+
+=== TEST 25: test remove ua-restriction part 4, check spider User-Agent
+--- request
+GET /hello
+--- more_headers
+User-Agent:Baiduspider/1.0
+--- response_body
+hello world
+
+
+
+=== TEST 26: set disable=true
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "/hello",
+                    "plugins": {
+                        "ua-restriction": {
+                            "denylist": [
+                                "foo"
+                            ],
+                            "disable": true
```

Comment:
What does `disable` mean here? I don't see it in the plugin properties doc.

Reply:
disable is built-in properties of plugin schema